sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/NoSuchItemException.scala

```diff
@@ -37,8 +37,12 @@ case class NoSuchNamespaceException(
     override val cause: Option[Throwable] = None)
   extends AnalysisException(message, cause = cause) {
 
+  def this(namespace: Array[String], cause: Option[Throwable]) = {
+    this(s"Namespace '${namespace.quoted}' not found", cause = cause)
+  }
+
   def this(namespace: Array[String]) = {
-    this(s"Namespace '${namespace.quoted}' not found")
+    this(namespace, cause = None)
```

Comment:
Why does this change `NoSuchNamespaceException`? Can't the caller add `cause = None`?

Reply:
+1, we should be able to do without.

Reply:
Actually, this is necessary for Java code to call `new NoSuchNamespaceException(namespace)` without a cause.

The following constructor (default value `None` to `cause`) will not work for Java:
```
  def this(namespace: Array[String], cause: Option[Throwable] = None)
```

How about having these 2 constructors?
```
  def this(namespace: Array[String], cause: Throwable)
  def this(namespace: Array[String])
```